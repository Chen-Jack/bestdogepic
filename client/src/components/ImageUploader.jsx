import React from 'react'
import { connect } from 'react-redux'
import { startUpload } from './../actions/upload'

export class ImageUploader extends React.Component {
  render () {
    return <div>
      <h1> Upload an Image! </h1>
      <input type='file' name='pic' accept='image/*' />
      <button type='button' onClick={this.props.startUpload}> Upload </button>
    </div>
  }
}

function mapDispatchToProps (dispatch, ownProps) {
  return {
    startUpload: () => dispatch(startUpload())
  }
}

export default connect(null, mapDispatchToProps)(ImageUploader)

