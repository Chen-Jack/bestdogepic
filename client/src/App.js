import React from 'react'
import logo from './logo.svg'
import { Provider } from 'react-redux'
import store from './store'
import './App.css'

import ImageUploader from './components/ImageUploader'

function App () {
  return (
    <Provider store={store} >
      <div className='App'>
        <header className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <ImageUploader />
        </header>
      </div>
    </Provider>
  )
}

export default App
