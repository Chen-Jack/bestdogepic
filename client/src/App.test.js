/* globals jest test expect describe */
import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'

import { ImageUploader } from './components/ImageUploader'

import { shallow, mount, render } from 'enzyme'

test('renders without crashing', () => {
  shallow(<App />)
})

describe('ImageUploader component', () => {
  test('Matches snapshot', () => {
    const wrapper = shallow(<ImageUploader />)
    expect(wrapper).toMatchSnapshot()
  })

  test('Has a file input field', () => {
    const wrapper = shallow(<ImageUploader />)
    expect(wrapper.exists('input[type="file"]')).toBe(true)
  })

  test('Should dispatch action', () => {
    const mockDispatch = jest.fn(() => {
      console.log('Sending data')
    })
    const wrapper = shallow(<ImageUploader startUpload={mockDispatch} />)
    expect(wrapper.exists('button')).toBe(true)

    const submitBtn = wrapper.find('button')
    submitBtn.simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})
