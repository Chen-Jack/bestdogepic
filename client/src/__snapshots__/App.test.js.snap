// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUploader component Matches snapshot 1`] = `
<div>
  <h1>
     Upload an Image! 
  </h1>
  <input
    accept="image/*"
    name="pic"
    type="file"
  />
  <button
    type="button"
  >
     Upload 
  </button>
</div>
`;
